### PR TITLE
[Bug fix] prevent squashing pair u8 dequantize -> s8 quantize

### PR DESCRIPTION
**Pull request by sfraczek**
### PR types
Bug fixes

### PR changes
Others

### Describe
* Added code that prevents squashing quantize->dequantize pairs if first is uint8 and second is int8. 
* Added fix for post training quantization checking for unsigned output. FC and Conv2d have different attribute names for fused activation.

### CODE REVIEW

**Comment by sfraczek** [comment]
please do not merge yet I want to add unit tests, there seems to be a perf regression

File: paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc [hunk 1]
```diff
@@ -104,6 +104,37 @@ void CPUQuantizeSquashPass::FindNodesToKeep(
...
+bool CPUQuantizeSquashPass::IsDequantizeInputUint8(
+    const Node* dequant_in) const {
+  PADDLE_ENFORCE_EQ(
+      dequant_in->inputs.size(), 1,
+      platform::errors::InvalidArgument(
+          "Dequantize (id: %f) should have only one input.", dequant_in->id()));
+  if (dequant_in->inputs[0]->IsOp()) {
+    auto prev_op = dequant_in->inputs[0]->Op();
+    std::string act_name;
+    std::cout << "prev_op->Type(): " << prev_op->Type() << std::endl;
```
Comment (baoachun) [inline paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc]: Please don't use `std::cout`.

**Comment by sfraczek** [comment]
I was comparing to wrong baseline. There shouldn't be a problem with this PR.

**Review by baoachun — APPROVED** [review]
LGTM

**Review by wozna — APPROVED** [review]
Great job, LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by baoachun** [comment]
Hi @sfraczek, QA found that this pr caused the `test_quant2_int8_lstm_mkldnn` single test execution to fail. Chould you please take a look?
https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/4797233/job/11924197
图片

**Comment by sfraczek** [comment]
Thank you very much for pointing this out, I will investigate this test. I am already looking into what this PR caused. I made one follow up fix to an error that was not caught by our unit tests in https://github.com/PaddlePaddle/Paddle/pull/39593.

**Comment by sfraczek** [comment]
I have reverted the change that caused this PTQ lstm problem on pr https://github.com/PaddlePaddle/Paddle/pull/39593.

**Pull request merged**